### PR TITLE
Don't fail if flakehub cache wasn't requested and its requirements weren't present

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: magic-nix-cache/src/main.rs [hunk 1]
```diff
@@ -244,21 +281,45 @@ async fn main_cli() -> Result<()> {
...
+
+            Some(FlakeHubAuthSource::DeterminateNixd)
         }
-        (Some(path), false) => FlakeHubAuthSource::Netrc(path),
-        (None, false) => {
+
+        // When determinate-nixd is not available, but the user specified a netrc
+        (_, Some(path), Dnixd::Missing) => Some(FlakeHubAuthSource::Netrc(path.to_owned())),
+
+        // User explicitly turned on flakehub cache, but it we have no netrc and determinate-nixd is not present
```
Comment (colemickens) [inline magic-nix-cache/src/main.rs:314]: nit, grammar

**Review by colemickens — APPROVED** [review]
tiny comment nit, but match checks out. important part is the `let flakehub_state = ` conditional change. makes sense, seems right.

File: magic-nix-cache/src/main.rs (hunk 1 above)
Comment (grahamc) [reply magic-nix-cache/src/main.rs:314]:
```suggestion
        // User explicitly turned on flakehub cache, but we have no netrc and determinate-nixd is not present
```

File: magic-nix-cache/src/main.rs [hunk 2]
```diff
@@ -129,6 +129,39 @@ struct Args {
...
+
+impl From<bool> for Dnixd {
+    fn from(b: bool) -> Self {
+        if b {
+            Dnixd::Available
+        } else {
+            Dnixd::Missing
+        }
+    }
+}
```
Comment (cole-h) [inline magic-nix-cache/src/main.rs:163]: This gives me pause... but I guess it's fine.

**Review by colemickens — APPROVED** [review]

**Pull request merged**